### PR TITLE
Improves the sql death report

### LINKED ISSUE #23413: More robust death reporting
Right now, deaths are reported with a lot of really useful information, but they're missing two very important pieces of data.

1. The server the death occurred on  
and  
2. The map file the death occurred on

Right now, the only way I can think of to determine which map a death occurred on would be to compare the coordinates against an array of areas, but that's a pain in the butt.

Ideally, the server/map info would go in a new column on the death table. Seeing as how SQL schema changes aren't easy to do, simply appending that data to the `ss13death.coord` column would work. Format's not super important, but if I got my wish, it'd look like:  
`(Xx,Yy,Zz)-MapName-ServerAddr`

EDIT: For the sake of backwards compatibility, the format should be:  
`Xx,Yy,Zz,MapName,ServerAddr`

### PR DESCRIPTION
**Pull request by lzimann**
As requested before, now the death report will also include the map name and server's address and port. 
I also removed the `sql_report_cyborg_death`, it was a copypasta of the proc above for some reason.

Closes #23413

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/death.dm [hunk 1]
```diff
@@ -30,4 +30,4 @@
 
 	update_icons()
 
-	sql_report_cyborg_death(src)
\ No newline at end of file
+	sql_report_death(src)
```
Comment (ChangelingRain) [inline code/modules/mob/living/silicon/robot/death.dm]: i'm pretty sure this will duplicate deaths for borgs though?

Comment (lzimann) [inline code/modules/mob/living/silicon/robot/death.dm]: How so?

Comment (ChangelingRain) [inline code/modules/mob/living/silicon/robot/death.dm]: well, it calls . = ..() so it should be inheriting the lower call

Comment (lzimann) [inline code/modules/mob/living/silicon/robot/death.dm]: Pretty sure this is only called in humans. Let me take a look.

**Comment by Jordie0608** [comment]
MSO will need to be around for this to be merged.
<sub><sub><sup><sup>If someone merges this before #23327 i'll stab them</sub></sub></sup></sup>

File: SQL/database_changelog.txt [hunk 2]
```diff
@@ -1,3 +1,13 @@
+30 January 2017, by Lzimann
+
+Modified table 'death', adding the columns 'mapname' and 'server'.
+
+ALTER TABLE `death` ADD COLUMN `mapname` TEXT NOT NULL AFTER `coord`, ADD COLUMN `server` TEXT NOT NULL AFTER `mapname`
```
Comment (MrStonedOne) [inline SQL/database_changelog.txt]: You can't just add a not null column and not provide a default for the existing rows.

add `DEFAULT ''` after NOT NULL or any server operator trying to import this will get errors they may not understand.

Comment (MrStonedOne) [inline SQL/database_changelog.txt]: wait, no, thats right, text can't have a default. one second, let me think

Comment (MrStonedOne) [inline SQL/database_changelog.txt]: Ya no i'm stupid, for text columns it just defaults to a blank string when adding the column. ignore this.

Comment (Jordie0608) [inline SQL/database_changelog.txt]: Have it be varchar then? I don't believe there'd be any real difference with the data we'd have.

Comment (MrStonedOne) [inline SQL/database_changelog.txt]: >TEXT and BLOB columns are implemented differently in the NDB storage engine, wherein each row in a TEXT column is made up of two separate parts. One of these is of fixed size (256 bytes), and is actually stored in the original table. The other consists of any data in excess of 256 bytes, which is stored in a hidden table. The rows in this second table are always 2,000 bytes long. This means that the size of a TEXT column is 256 if size <= 256 (where size represents the size of the row); otherwise, the size is 256 + size + (2000 − (size − 256) % 2000).

So basically we really should be using varchar for all of these.

**Comment by MrStonedOne** [comment]
I added the columns

**Comment by Jordie0608** [comment]
lzi is a nerd

**Comment by lzimann** [comment]
no u
